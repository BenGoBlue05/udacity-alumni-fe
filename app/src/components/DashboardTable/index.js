import React, { PropTypes } from 'react';
import Box from 'grommet-udacity/components/Box';
import Table from 'grommet-udacity/components/Table';
import TableRow from 'grommet-udacity/components/TableRow';
import Anchor from 'grommet-udacity/components/Anchor';

const DashboardTable = ({
  articles,
}) => (
  <Box
    pad="large"
    style={{ minHeight: 300, minWidth: 600, border: '1px solid black' }}
    color="light-2"
  >
    <Table>
      <thead>
        <tr>
          <th>Title</th>
          <th>Status</th>
          <th>Author</th>
          <th>Actions</th>
        </tr>
      </thead>
      <tbody>
        {articles.map((article, i) =>
          <TableRow key={i}>
            <td>
              <Anchor href={`/articles/${article.id}`}>
                {article.title.slice(0, 15)}
              </Anchor>
            </td>
            <td>
              <Anchor href={`/admin/cms/${article.id}`}>
                {`${article.status.charAt(0).toUpperCase()}${article.status.slice(1)}`}
              </Anchor>
            </td>
            <td>
              <Anchor href={`/admin/cms/${article.id}`}>
                {article.user.name}
              </Anchor>
            </td>
            <td>
              <span>
                <Anchor href={`/admin/cms/${article.id}/edit`}>
                  Edit
                </Anchor>
                {' / '}
                <Anchor href={`/admin/cms/${article.id}/delete`}>
                  Delete
                </Anchor>
              </span>
            </td>
          </TableRow>
        )}
      </tbody>
    </Table>
  </Box>
);

DashboardTable.propTypes = {
  articles: PropTypes.array.isRequired,
};

export default DashboardTable;
