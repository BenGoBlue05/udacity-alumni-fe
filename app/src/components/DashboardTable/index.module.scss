.boxWrapper {
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-direction: row !important;
}

.imageWrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: row !important;
}

.innerWrapper {
  width: 80vw;
  justify-content: center;
}

.listWrapper {
  min-height: 750px;
  max-width: 100vw !important;
  box-sizing: border-box;
}

.flexGrow {
  flex-grow: 1;
}

.thumbnailWrapper {
  width: 100px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.thumbnail {
  max-width: 50px !important;
  display: inline-block;
  padding: 4px;
  line-height: 1.42857143;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 4px;
  transition: all .2s ease-in-out;
}
