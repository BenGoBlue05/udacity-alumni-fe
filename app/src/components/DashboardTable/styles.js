import styled from 'styled-components';
import GrommetTableRow from 'grommet-udacity/components/TableRow';
import Box from 'grommet-udacity/components/Box';
import Heading from 'grommet-udacity/components/Heading';

export const TableRow = styled(GrommetTableRow)`
  cursor: pointer;
  &:hover {
    background-color: #eee;
  }
  border-bottom: 1px solid #eee;
`;

export const ArticleHeadingWrapper = styled(Box)`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
`;

export const ArticleHeading = styled(Heading)`
  margin-top: 30px;
`;
