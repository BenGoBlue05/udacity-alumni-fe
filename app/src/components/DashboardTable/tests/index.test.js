import DashboardTable from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<DashboardTable />', () => {
  it('should render with default props', () => {
    const articles = [
      {
        id: 0,
        status: 'published',
        title: 'Hello World',
        user: {
          name: 'Ryan Collins',
        },
      },
    ];
    const wrapper = shallow(
      <DashboardTable articles={articles} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
