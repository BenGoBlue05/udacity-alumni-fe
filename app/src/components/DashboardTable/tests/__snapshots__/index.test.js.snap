exports[`<DashboardTable /> should render with default props 1`] = `
<Box
  announce={false}
  color="light-2"
  direction="column"
  focusable={true}
  pad="large"
  responsive={true}
  tag="div">
  <Table>
    <thead>
      <tr>
        <th>
          Title
        </th>
        <th>
          Status
        </th>
        <th>
          Author
        </th>
        <th>
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <TableRow>
        <td>
          <Heading
            tag="h4">
            Hello World
          </Heading>
        </td>
        <td>
          <Heading
            tag="h5">
            Published
          </Heading>
        </td>
        <td>
          <Heading
            tag="h5">
            Ryan Collins
          </Heading>
        </td>
        <td>
          <DashboardTableButtonMenu
            article={
              Object {
                "id": 0,
                "status": "published",
                "title": "Hello World",
                "user": Object {
                  "name": "Ryan Collins"
                }
              }
            }
            onDeleteArticle={[Function onDeleteArticle]} />
        </td>
      </TableRow>
    </tbody>
  </Table>
</Box>
`;
